plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMaterializedViewProperties.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.spi.session.PropertyMetadata;
+
+import javax.inject.Inject;
+
+import java.util.List;
+
+public class IcebergMaterializedViewProperties
+{
+    private final List<PropertyMetadata<?>> materializedViewProperties;
+
+    @Inject
+    public IcebergMaterializedViewProperties(IcebergTableProperties storageTableProperties)
+    {
+        materializedViewProperties = ImmutableList.<PropertyMetadata<?>>builder()
+                // Materialized view should allow configuring all the supported iceberg table properties for the storage table
+                .addAll(storageTableProperties.getTableProperties())
```

Comment:
Source of truth for table properties is in `IcebergConnector`.
For example, we could switch it to use `TablePropertiesProvider`.

Rename IcebergMaterializedViewProperties to IcebergMaterializedViewAdditionalProperties 
and let the `IcebergConnector` union the two sets in IcebergConnector ctor

Reply:
Made this change. I've squashed the initial commit now as it looked very awkward to have a preparatory commit with empty properties.